ParserTests.fs

```diff
@@ -4,127 +4,164 @@
-    member x.``the atomic set parser parses sets correctly`` expr =
-        expr
-        |> run parseAtomicSet
-        |> ParserTests.ParseResultToOptional
-
-
-    /// Test cases for testing the constraint parser.
-    static member ConstraintParses =
-        [ TestCaseData("constraint emp -> true;")
-              .Returns(Some (Definite (DView.Unit, True)))
-          TestCaseData("constraint Func(a, b) -> c > a + b;")
-              .Returns(Some (Definite
-                                 (DView.Func { Name = "Func"
-                                               Params = [ "a"; "b" ] },
-                                  Bop (Gt,
-                                       LV(LVIdent "c"),
-                                       Bop(Add,
-                                           LV(LVIdent "a"),
-                                           LV(LVIdent "b"))))))
-          TestCaseData("constraint Func(a, b) -> ?;")
-              .Returns(Some (Indefinite
-                                 (DView.Func { Name = "Func"
-                                               Params = [ "a"; "b" ] } )
-                             : ViewDef<DView, Expression>))
-          TestCaseData("constraint Func(a, b) -> %{uninterpreted symbol}() == true;")
-              .Returns(Some (Definite
-                                 (DView.Func { Name = "Func"
-                                               Params = [ "a"; "b" ] },
-                                  Bop (Eq,
-                                       Symbolic("uninterpreted symbol", []),
-                                       True))
-                             : ViewDef<DView, Expression>)) ]
-        |> List.map (fun d -> d.SetName(sprintf "Parse %A" d.OriginalArguments.[0]))
-
-    [<TestCaseSource("ConstraintParses")>]
-    /// Tests whether the constraint parser works correctly.
-    member x.``the constraint parser parses valid constraints correctly`` cs =
-        cs
-        |> run parseConstraint
-        |> ParserTests.ParseResultToOptional
-
-    /// Test cases for testing the full parser.
-    static member ScriptParses =
-        [ TestCaseData(Starling.Tests.Studies.ticketLock).Returns(Some(Starling.Tests.Studies.ticketLockParsed))
-            .SetName("Parse the ticket lock") ]
-
-    [<TestCaseSource("ScriptParses")>]
-    /// Tests whether the script parser works correctly.
-    member x.``the script parser parses full case studies correctly`` expr =
-        expr
-        |> run parseScript
-        |> ParserTests.ParseResultToOptional
+open FParsec
+
+let get =
+    function
+    | Success(r, _, _) -> Some r
+    | Failure _ -> None
+
+let check p str ast = 
+    let actual = run p str 
+    Assert.AreEqual (ast, get actual)
+
+
+// Perform the same trick matt uses in Main.fs to overwrite a right-associative
+// operator with the correct behaviour
+let ( ** ) = ( <| )
+
+
+// Conversion of mattw's test cases into new system
+module ExpressionTests = 
+    [<Test>]
```

Comment:
I'd probably pick either `TestCaseData` or `Test` and use it everywhere.  I'm beginning to regret `TestCaseData` as it doesn't work well with F# type inference, if that helps :P